python/ray/tune/tune.py

```diff
@@ -341,8 +342,26 @@ def run(
     if num_samples == -1:
         num_samples = sys.maxsize
 
+    result_buffer_length = None
+    if isinstance(scheduler, HyperBandScheduler):
+        # Result buffering with a Hyperband scheduler is a bad idea, as
+        # hyperband tries to stop trials when processing brackets. With result
+        # buffering, we might trigger this multiple times when evaluating
+        # a single trial, which leads to unexpected behavior.
+        env_result_buffer_length = os.getenv("TUNE_RESULT_BUFFER_LENGTH", "")
+        if env_result_buffer_length:
+            warnings.warn(
+                f"You are using a Hypberband scheduler, but "
+                f"TUNE_RESULT_BUFFER_LENGTH is set "
+                f"({env_result_buffer_length}). This can lead to undesired "
+                f"and faulty behavior, so the buffer length was foracbly set "
+                f"to 1 instead.")
```

Comment:
```suggestion
                "You are using a Hyperband scheduler, but "
                "TUNE_RESULT_BUFFER_LENGTH is set "
                f"({env_result_buffer_length}). This can lead to undesired "
                "and faulty behavior, so the buffer length was forcibly set "
                "to 1 instead.")
```